syft/frameworks/torch/mpc/falcon/falcon_helper.py

```diff
@@ -0,0 +1,135 @@
+    def __private_unfold(image, kernel_size, padding):
+        return image.unfold(kernel_size, padding)
+
+    @staticmethod
+    def xor(
+        value: Union[int, ReplicatedSharingTensor], other: Union[int, ReplicatedSharingTensor]
+    ) -> ReplicatedSharingTensor:
+        """
+        Compute the XOR value between value and other.
+        If value and other are both ints we should use the "^" operator.
+
+        Args:
+            value (ReplicatedSharingTensor or Int): RST with ring size of 2 or Int value in {0, 1}
+            other (ReplicatedSharingTensor or Int): RST with ring size of 2 or Int value in {0, 1}
+
+        Returns:
+            The XOR computation between value and other
+        """
+
+        if not any(isinstance(x, ReplicatedSharingTensor) for x in (value, other)):
+            raise ValueError("One of the arguments should be a RST")
+
+        if all((isinstance(x, ReplicatedSharingTensor) for x in (value, other))) and not all(
+            (x.ring_size == FalconHelper.XOR_EXPECTED_RING_SIZE for x in (value, other))
+        ):
+            raise ValueError("If both arguments are RST then they should be in ring size 2")
+
+        # Make sure we have a ReplicatedSharingTensor as value
+        if isinstance(value, int):
+            if 0 <= value <= 1:
+                return FalconHelper.xor(other, value)
+            else:
+                raise ValueError("The integer value should be in {0, 1}")
+
+        return value + other - (value * 2 * other)
+
+    @staticmethod
+    def __switch_public_private(value, public_function, private_function, *args, **kwargs):
+        if isinstance(value, (int, float, torch.Tensor, syft.FixedPrecisionTensor)):
+            return public_function(value, *args, **kwargs)
+        elif isinstance(value, syft.ReplicatedSharingTensor):
+            return private_function(value, *args, **kwargs)
+        else:
+            raise ValueError(
+                "expected int, float, torch tensor, or ReplicatedSharingTensor "
+                "but got {}".format(type(value))
+            )
+
+    @staticmethod
+    def select_share(
+        b: ReplicatedSharingTensor,
+        x: ReplicatedSharingTensor,
+        y: ReplicatedSharingTensor,
+    ) -> ReplicatedSharingTensor:
+        """Select x or y depending on b
+
+        Args:
+            x (ReplicatedSharingTensor): RST that will be selected if b reconstructed is 0
+            y (ReplicatedSharingTensor): RST that will be selected if b reconstructed is 1
+            b (ReplicatedSharingTensor): RST of a bit
+
+        Return:
+            x if b == 0 else y
+        """
+
+        ring_size = x.ring_size
+        players = x.players
+
+        # This should be generated by a crypto provider or create a 0 RST shared value and make
+        # each worker add a random number at an offline step
+        # (Idea): Use the correlated randomness shared at the beginning
+        c = torch.randint(high=2, size=(1,))
```

Comment:
should the size be 1 or x.shape?

Reply:
Reading that part of the paper again - yeah it should be ```x.shape```